presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -156,4 +157,14 @@ public Expression rewriteExtract(Extract node, Void context, ExpressionTreeRewri
             throw new UnsupportedOperationException("not yet implemented: " + node.getField());
         }
     }
+
+    private static boolean isLiteral(Expression expression)
```

Comment:
`isConstant`?

Reply:
Sure. Although isConstant implies more than what this is. It's not about arbitrary constant expressions, but AST-encoded typed literals.